File: frontend/src/types/specialty.ts
export interface Specialty {
  id: string;
  name: string;
  slug: string;
  description: string;
  imageUrl: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateSpecialtyDto {
  name: string;
  description: string;
  imageUrl: string | null;
}

export interface SpecialtyInfo {
  specialtySlug: string;
  specialtyName: string;
  specialtyDescription: string;
  specialtyImage?: string;
}

export interface UpdateSpecialtyDto extends CreateSpecialtyDto {}

File: backend/Models/Specialty.cs
namespace backend.Models
{
    public class Specialty
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public string Slug { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string? ImageUrl { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public virtual ICollection<DoctorProfile> Doctors { get; set; } = new List<DoctorProfile>();
    }
}
